server/src/main/java/com/cloud/network/element/VpcVirtualRouterElement.java

```diff
@@ -348,7 +348,7 @@ public Provider getProvider() {
 
         //1st time it runs the domain router of the VM shall be returned
         List<DomainRouterVO> routers = super.getRouters(network, dest);
-        if (routers.size() > 0) {
+        if (router != null && routers.size() > 0) {
```

Comment:
@ravening typo here, pls correct. packaging failed due to this.

```suggestion
        if (routers != null && routers.size() > 0) {
```

Reply:
@sureshanaparti  done